pkg/apis/extensions/v1alpha1/types_bastion.go

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ Object = (*Bastion)(nil)
+
+// BastionResource is a constant for the name of the Bastion resource.
+const BastionResource = "Bastion"
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:scope=Namespaced,path=bastions,singular=bastion
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name=IP,JSONPath=".spec.bastionIP",type=string,description="The public IP address of the temporary bastion host"
+// +kubebuilder:printcolumn:name=State,JSONPath=".status.state",type=string,description="The bastion's status."
+// +kubebuilder:printcolumn:name=Age,JSONPath=".metadata.creationTimestamp",type=date,description="creation timestamp"
+
+// Bastion is a bastion or jump host that is dynamically created
+// to provide SSH access to shoot nodes.
+type Bastion struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object metadata.
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
```

Comment:
I think you can remove all the `protobuf` tags, we don't generate this for the CRDs, only for the API groups served by GAPI (gardener-apiserver).

Reply:
`make generate-extensions-crds` exists for what reason then?

And yes, I was wondering how the extension CRDs move from this repo to the extensions-* repos. The AWS one does have example CRDs, but they are not spec'ed out and seem handwritten.

Reply:
Do you want me to put the new CRD YAML file into the chart or is this done at a later point in time?

Reply:
I think it makes sense to add it as part of this PR because it adds the APIs as well 👍🏻

Reply:
Hmm when generating the CRDs, I get them as `apiextensions/v1` already, whereas all other CRDs in that chart are still v1beta1. I therefore tried to just regenerate all CRDs and copy them over, but a few, like that for ManagedResources, was not included in the extensions group. So I'm not sure if we want to have one oddball CRD or if I messed up somehow.

Also, I manually added the `gardener.cloud/deletion-protected: "true"` label to the CRD, assuming all extensions should have it set.

Reply:
@rfranzke How do we want to solve this? :)

Reply:
I fear that we still need the CRDs in `v1beta1` because we still support Kubernetes versions that don't have the `v1` API of the CRDs yet. 

> Also, I manually added the `gardener.cloud/deletion-protected: "true"` label to the CRD, assuming all extensions should have it set.

Sounds good 👍🏻